### ISSUE TITLE
Salesforce with dj-rest-auth: 400 error, "Incorrect value"

### ISSUE THREAD

**Issue by sunweiyang** [jazzband/dj-rest-auth#184]
My mobile and web frontends can already sign in using Facebook and Google via dj-rest-auth. I'm now trying to add Salesforce as a 3rd REST social login method, but am running into issues.

I've followed the [django-allauth instructions for Salesforce](https://django-allauth.readthedocs.io/en/latest/providers.html#salesforce):

* Created a Salesforce Connected App with id and openid scopes
* Created a SocialApplication in Django with client ID, secret, and login URL in the "Key" field (`https://login.salesforce.com/`)
* Included allauth.socialaccount.providers.salesforce in INSTALLED_APPS

Running `jsforce.browser.login()` in my clients' JS code opens a Salesforce login popup. After entering Salesforce login credentials, the Salesforce system redirected to my defined callback URL, which included a hash containing an access token looking something like this:

```
https://www.mycallback.com#access_token=00D3t000004QWRm%21ARwAQPfHWiM6jdB43dlyW6qjEw._34mjzGi_Jv6YCXp0QssT.9F9lCge5_YaH8gqTy3Od6SywCs8X9zOGv145SyviBVeGdn0&instance_url=https%3A%2F%2Fna123.salesforce.com&id=https%3A%2F%2Flogin.salesforce.com%2Fid%2F00D3t000004QWRmEAO%2F0053t000008QBetAAG&issued_at=1606802917608&signature=KvxAX0WBCFQYY%2BO25id9%2FXxpbh2q2d2vWdQ%2FFV5FCBw%3D&state=jsforce0.popup.c0ockgct29g&scope=id+openid&token_type=Bearer
```

I tried sending the `access_token` from that URL's hash to my Salesforce API endpoint (see below), but that resulted in a 400 error ("Incorrect value").

Here is how I've defined my `SocialLoginView` classes in my views.py:

```python
from dj_rest_auth.registration.views import SocialLoginView
from allauth.socialaccount.providers.facebook.views import FacebookOAuth2Adapter
from allauth.socialaccount.providers.google.views import GoogleOAuth2Adapter
from allauth.socialaccount.providers.salesforce.views import SalesforceOAuth2Adapter


class FacebookLogin(SocialLoginView):
    adapter_class = FacebookOAuth2Adapter


class GoogleLogin(SocialLoginView):
    adapter_class = GoogleOAuth2Adapter


class SalesforceLogin(SocialLoginView):
    adapter_class = SalesforceOAuth2Adapter
```

My urls.py:

```python
from .views import FacebookLogin, GoogleLogin, SalesforceLogin

urlpatterns = [
    ...
    # Sending access_token to the Facebook and Google REST endpoints works,
    # but doing the same for the Salesforce REST endpoint does not (400 error: "Incorrect value")
    url(r'^api/rest-auth/facebook/$', FacebookLogin.as_view(), name='fb_login'),
    url(r'^api/rest-auth/google/$', GoogleLogin.as_view(), name='google_login'),
    url(r'^api/rest-auth/salesforce/$', SalesforceLogin.as_view(), name='salesforce_login'),
    ...
]
```

**Comment by UsmanJ** [comment]
Did you manage to figure out why this was happening? I am having a similar issue.

**Comment by sunweiyang** [comment]
@UsmanJ I figured it out and got it working: when posting to my dj-rest-auth Salesforce API endpoint, I was only including `access_token` in my POST body. I actually need both `access_token` and `key`, where key is the Salesforce login URL ("https://login.salesforce.com").

It was indeed in the django-allauth instructions for Salesforce, but I misinterpreted the wording. I now know that it says to require both `access_token` and `key` in the POST body.
